progenitor-impl/src/cli.rs

```diff
@@ -0,0 +1,552 @@
+// Copyright 2023 Oxide Computer Company
+
+use heck::ToKebabCase;
+use openapiv3::OpenAPI;
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote};
+use typify::{TypeSpaceImpl, TypeStructPropInfo};
+
+use crate::{
+    method::{
+        OperationParameterKind, OperationParameterType, OperationResponseStatus,
+    },
+    space_out_items,
+    to_schema::ToSchema,
+    util::{sanitize, Case},
+    validate_openapi, Generator, Result,
+};
+
+struct CliOperation {
+    cli_fn: TokenStream,
+    execute_fn: TokenStream,
+    cli_trait: TokenStream,
+    execute_trait: TokenStream,
+}
+
+impl Generator {
+    pub fn cli_text(
+        &mut self,
+        spec: &OpenAPI,
+        crate_name: &str,
+    ) -> Result<String> {
+        let output = self.cli(spec, crate_name)?;
+
+        let content = rustfmt_wrapper::rustfmt_config(
```

Comment:
according to https://github.com/oxidecomputer/progenitor/issues/362 , wouldnt `rustfmt` not be used in the -impl ?

Reply:
Good point; we'll work this out.